web/src/classic/components/organisms/EarthEditor/Header/hooks.ts

```diff
@@ -80,9 +81,16 @@ export default () => {
     },
     [checkProjectAliasQuery, project],
   );
+
   useEffect(() => {
-    if (!currentWorkspace && lastWorkspace) setWorkspace(lastWorkspace);
-  }, [currentWorkspace, lastWorkspace, setWorkspace]);
+    if (!currentWorkspace && lastWorkspace && workspaceId == lastWorkspace.id)
+      setWorkspace(lastWorkspace);
+    else {
+      const workspace = workspaces?.find(workspace => workspace.id === workspaceId);
+      setWorkspace(workspace);
+      setLastWorkspace(workspace);
+    }
+  }, [currentWorkspace, lastWorkspace, setLastWorkspace, setWorkspace, workspaceId, workspaces]);
```

Comment:
```suggestion
  if (currentWorkspace || !lastWorkspace || workspaceId !== lastWorkspace.id) {
    const workspace = workspaces?.find((workspace) => workspace.id === workspaceId);
    setWorkspace(workspace);
    setLastWorkspace(workspace);
  } else {
    setWorkspace(lastWorkspace);
  }
}, [currentWorkspace, lastWorkspace, setLastWorkspace, setWorkspace, workspaceId, workspaces]);
```

Reply:
sorry dear I don't think this condition is right , for example assume that we don't have current workspace that doesn't mean that we are sure that we have last workspace